.github/workflows/GCC.yml

```diff
@@ -58,6 +53,36 @@ jobs:
         spack config add "packages:mpi:require:'${{ matrix.mpi }}'"
         spack concretize |& tee ${SPACK_ENV}/log.concretize
         spack install -j2 --fail-fast
+        spack clean --all
+
+  build_fv3atm:
+    needs: build_spack
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        cmake_opts: ["-D32BIT=ON", "-D32BIT=OFF"]
+        gcc_ver: ["12"]
+        mpi: ["mpich", "openmpi"]
+
+    steps:
+
+    - name: install-doxygen
+      run: |
+        sudo apt-get install doxygen graphviz
+
+    - name: checkout-fv3atm
+      uses: actions/checkout@v4
+      with:
+        path: ${{ github.workspace }}/fv3atm
+        submodules: recursive
+
+    - name: cache-spack
+      id: cache-spack
```

Comment:
Since the Spack cache isn't/shouldn't be modified in the part of the workflow, and since the cache should definitely exist at this point, I think we could use `uses: actions/cache/restore@v4` (https://github.com/actions/cache).

Reply:
action/cache/restore works for build_fv3atm job. Thanks for suggestion.